src/endpoints/minidump.rs

```diff
@@ -162,3 +162,109 @@ pub fn configure(app: ServiceApp) -> ServiceApp {
         r.method(Method::POST).with(handle_minidump_request);
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use std::fs;
+
+    use actix_web::test::TestServer;
+    use reqwest::{multipart, Client, StatusCode};
+    use tokio::runtime::Runtime;
+
+    use crate::app::ServiceState;
+    use crate::config::Config;
+    use crate::test;
+    use crate::types::SymbolicationResponse;
+
+    #[tokio::test]
+    async fn test_basic() {
+        test::setup();
+
+        let service = ServiceState::create(Config::default()).unwrap();
+        let server = TestServer::with_factory(move || crate::server::create_app(service.clone()));
+
+        let file_contents = fs::read("tests/fixtures/windows.dmp").unwrap();
+        let file_part = multipart::Part::bytes(file_contents).file_name("windows.dmp");
+
+        let form = multipart::Form::new()
+            .part("upload_file_minidump", file_part)
+            .text("sources", "[]");
+
+        let response = Client::new()
+            .post(&server.url("/minidump"))
+            .multipart(form)
+            .send()
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let body = response.text().await.unwrap();
+        let response = serde_json::from_str::<SymbolicationResponse>(&body).unwrap();
+        insta::assert_yaml_snapshot!(response);
+    }
+
+    #[test]
+    fn test_integration_microsoft() {
+        // TODO: Move this test to E2E tests
+        test::setup();
+
+        // XXX: This cannot be a `#[tokio::test]` because the service spawns a tokio runtime
+        // internally. Dropping this within another runtime created by `tokio::test` is not allowed.
+        let service = ServiceState::create(Config::default()).unwrap();
```

Comment:
I still don't get why this doesn't apply to the other tests?

Reply:
Neither do I. My best guess is that either the runtime initializes lazily, or if no futures are spawned in the runtime its `Drop` does not acquire certain resources that lead to the drop.